integration_tests/src/main/scala/org/apache/spark/sql/tests/datagen/big_data_gen.scala

```diff
@@ -0,0 +1,1923 @@
+    if (lowerName.contains(".")) {
+      // Not sure if there are other forbidden characters, but we can check on that later
+      throw new IllegalArgumentException("Name cannot contain '.' character")
+    }
+    if (tables.contains(lowerName)) {
+      throw new IllegalArgumentException("Cannot add duplicate tables (even if case is different)")
+    }
+    val ret = new TableGen(columns, numRows)
+    tables.put(lowerName, ret)
+    ret
+  }
+
+  /**
+   * Add a new table with a given type
+   * @param name the name of the table (must be unique)
+   * @param st the type for this table.
+   * @param numRows the number of rows for this table
+   * @return the TableGen that was just added
+   */
+  def addTable(name: String, st: StructType, numRows: Long): TableGen = {
+    val localTableId = tableId
+    tableId += 1
+    val mapping = OrderedTypeMapping(mappings.toArray)
+    val sg = DBGen.dtToTopLevelGen(st, localTableId, defaultRanges, numRows, mapping)
+    addTable(name, sg, numRows)
+  }
+
+  /**
+   * Add a new table with a type defined by the DDL
+   * @param name the name of the table (must be unique)
+   * @param ddl the DDL that describes the type for the table.
+   * @param numRows the number of rows for this table
+   * @return the TableGen that was just created
+   */
+  def addTable(name: String, ddl: String, numRows: Long): TableGen = {
+    val localTableId = tableId
+    tableId += 1
+    val mapping = OrderedTypeMapping(mappings.toArray)
+    val sg = DBGen.dtToTopLevelGen(
+      DataType.fromDDL(ddl).asInstanceOf[StructType], localTableId, defaultRanges, numRows, mapping)
+    addTable(name, sg, numRows)
+  }
+
+  /**
+   * Get a TableGen by name
+   * @param name the name of the table to look for
+   * @return the corresponding TableGen
+   */
+  def apply(name: String): TableGen = tables(name.toLowerCase)
+
+  /**
+   * Names of the tables.
+   */
+  def tableNames: Seq[String] = tables.keys.toSeq
+
+  /**
+   * Take all of the tables and create or replace temp views with them using the given name.
+   * The data will be generated inline as the query runs.
+   * @param spark the session to use
+   * @param numParts the number of parts to use, if > 0
+   */
+  def createOrReplaceTempViews(spark: SparkSession, numParts: Int = 0): Unit = {
+    tables.foreach {
+      case (name, gen) =>
+        gen.toDF(spark, numParts).createOrReplaceTempView(name)
+    }
+  }
+
+  /**
+   * Write all of the tables out as parquet under path/table_name and overwrite anything that is
+   * already there.
+   * @param spark the session to use
+   * @param path the base path to write the data under
+   * @param numParts the number of parts to use, if > 0
+   */
+  def writeParquet(spark: SparkSession, path: String, numParts: Int = 0): Unit = {
+    tables.foreach {
+      case (name, gen) =>
+        val subPath = path + "/" + name
+        gen.toDF(spark, numParts).write.mode("overwrite").parquet(subPath)
```

Comment:
Do we want this to be dangerous by default?  Wondering if we need to worry about accidental data loss due to overwrite.